### ISSUE TITLE
docs(header): header have buttons without accessible names #3518 

### ISSUE THREAD

**Issue by bdryanovski** [clr-team/planning#94]
## Description of work

Header documentation was missing aria-label for icons

## Issue(s)

* https://github.com/vmware/clarity/issues/3518

## Definition of done

* [x] Tests completed
* [ ] Reviewed by peers
* [ ] Design sign off (if necessary)
* [ ] Merged into master/topic branch
* [ ] Backport (if necessary)
